script.module.simplecache/lib/simplecache.py

```diff
@@ -0,0 +1,311 @@
+            return
+        self._win.setProperty("simplecachecleanbusy", "busy")
+
+        query = "SELECT id, expires FROM simplecache"
+        for cache_data in self._execute_sql(query).fetchall():
+            cache_id = cache_data[0]
+            cache_expires = cache_data[1]
+            if self._exit or self._monitor.abortRequested():
+                return
+            # always cleanup all memory objects on each interval
+            self._win.clearProperty(cache_id.encode("utf-8"))
+            # clean up db cache object only if expired
+            if cache_expires < cur_timestamp:
+                query = 'DELETE FROM simplecache WHERE id = ?'
+                self._execute_sql(query, (cache_id,))
+                self._log_msg("delete from db %s" % cache_id)
+
+        # compact db
+        self._execute_sql("VACUUM")
+
+        # remove task from list
+        self._busy_tasks.remove(__name__)
+        self._win.setProperty("simplecache.clean.lastexecuted", repr(cur_time))
+        self._win.clearProperty("simplecachecleanbusy")
+        self._log_msg("Auto cleanup done")
+
+    def _get_database(self):
+        '''get reference to our sqllite _database - performs basic integrity check'''
+        addon = xbmcaddon.Addon(ADDON_ID)
+        dbpath = addon.getAddonInfo('profile')
+
+        try:
+            dbfile = xbmc.translatePath("%s/simplecache.db" % dbpath).decode('utf-8')
+        except AttributeError: # Python 3 compatiblity
+            dbfile = xbmc.translatePath("%s/simplecache.db" % dbpath)
+
+        if not xbmcvfs.exists(dbpath):
+            xbmcvfs.mkdirs(dbpath)
+        del addon
+        try:
+            connection = sqlite3.connect(dbfile, timeout=30, isolation_level=None)
+            connection.execute('SELECT * FROM simplecache LIMIT 1')
+            return connection
+        except Exception as error:
+            # our _database is corrupt or doesn't exist yet, we simply try to recreate it
+            if xbmcvfs.exists(dbfile):
+                xbmcvfs.delete(dbfile)
+            try:
+                connection = sqlite3.connect(dbfile, timeout=30, isolation_level=None)
+                connection.execute(
+                    """CREATE TABLE IF NOT EXISTS simplecache(
+                    id TEXT UNIQUE, expires INTEGER, data TEXT, checksum INTEGER)""")
+                return connection
+            except Exception as error:
+                self._log_msg("Exception while initializing _database: %s" % str(error), xbmc.LOGWARNING)
+                self.close()
+                return None
+
+    def _execute_sql(self, query, data=None):
+        '''little wrapper around execute and executemany to just retry a db command if db is locked'''
+        retries = 0
+        result = None
+        error = None
+        # always use new db object because we need to be sure that data is available for other simplecache instances
+        with self._get_database() as _database:
```

Comment:
Check for abortrequested

Reply:
I guess you mean in the next line in the while loop. If yes, done.

Reply:
Yes, sorry did the review on the phone